### DISCUSSION TITLE
redash server high availability

### DISCUSSION THREAD

**Discussion by anujthedevopsguy** [getredash/redash#6766] [Q&A]
I have setup a Redash EC2 instance and set auto-scaling to 1. Currently, Redash has user accounts, a dashboard, queries, and other data. Let's say that if this instance goes down, all this data will be lost. A new redash instance will be created due to auto-scaling, though. How to avoid losing data.

**Comment by justinclift** [comment]
You need to backup your PostgreSQL database, and the environment file (`env`) that's in the `/opt/redash` directory.

With those, you're able to recreate your Redash instance if you need to for some reason. :smile:

**Comment by anujthedevopsguy** [comment]
Are you referring the DB which we use as data source to produce dashboard?
Or db instance running into redash EC2 instance? I am new to redash.
questions could be vague :). where is created users info, dashboard ,
queries data saved ?

**Reply by justinclift** [reply]
Sorry, I wasn't as clear as I should have been.

Redash internally uses a PostgreSQL database that it stores all of the info (data sources, user info, queries, etc) in.

That's what you need to back up, and also make a copy of the matching environment file for.

I'm pretty sure we have instructions written somewhere for how to do the backup, but I'm not finding them in some quick searching.  Hopefully one of our other Community members will be able to provide the link... :smile:
